src/stdlib_io_disp.fypp

```diff
@@ -0,0 +1,181 @@
+    end procedure disp_0_${type[0]}$${kind}$
+
+    module procedure disp_1_${type[0]}$${kind}$
+        !! Display a/an ${type}$ vector.
+        integer :: unit_
+        logical :: brief_
+        integer :: m
+        unit_  = merge(unit, output_unit, present(unit))
+        brief_ = merge(brief, .false., present(brief))
+        m = size(value, 1)
+        if(present(header)) write(unit_, *) header
+        write(unit_, *) '[vector size: '//to_string(m)//']'
+        if(brief_ .and. m > 5) then
+            !! Brief Print.
+            write(unit_, fmt_r) value(1:3), '...', value(m)
+        else
+            !! Full Print.
+            write(unit_, fmt_r) value(:)
+        end if
+    end procedure disp_1_${type[0]}$${kind}$
+
+    module procedure disp_2_${type[0]}$${kind}$
+        !! Display a/an ${type}$ matrix.
+        integer :: unit_
+        logical :: brief_
+        character(1) :: colon(5)
+        integer :: i, m, n
+        unit_  = merge(unit, output_unit, present(unit))
+        brief_ = merge(brief, .false., present(brief))
+        m = size(value, 1)
+        n = size(value, 2)
+
+        if(present(header)) write(unit_, *) header
+        write(unit_, *) '[matrix size: '//to_string(m)//'×'//to_string(n)//']'
+        if(brief_ .and. (m > 5 .or. n > 5)) then
+            !! Brief Print.
+            if(m > 5 .and. n > 5) then
+                do i = 1, 3
+                    write(unit_, fmt_r) value(i,1:3), '...', value(i,n)
+                end do
+                write(unit_, fmt_r) ':', ':', ':', ':', ':'
+                write(unit_, fmt_r) value(m,1:3), '...', value(m,n)
+            elseif(m > 5 .and. n <= 5) then
+                do i = 1, 3
+                    write(unit_, fmt_r) value(i,:)
+                end do
+                colon(1:n) = ':'
+                write(unit_, fmt_r) colon(1:n)
+                write(unit_, fmt_r) value(m,:)
+            elseif(m <= 5 .and. n > 5) then
+                do i = 1, m
+                    write(unit_, fmt_r) value(i,1:3), '...', value(i,n)
+                end do
+            end if
+        else
+            !! Full Print.
+            do i = 1, m
+                write(unit_, fmt_r) value(i,:)
+            end do
+        end if
+    end procedure disp_2_${type[0]}$${kind}$
+    #:endfor
+
+    #:for kind, type in CMPLX_KINDS_TYPES
+    module procedure disp_0_${type[0]}$${kind}$
+        !! Display a ${type}$ scalar.
+        integer :: unit_
+        unit_  = merge(unit, output_unit, present(unit))
+        if(present(header)) write(unit_, *) header
+        write(unit_, fmt_c) format_to_string(value, format_)
+    end procedure disp_0_${type[0]}$${kind}$
+
+    module procedure disp_1_${type[0]}$${kind}$
+        !! Display a ${type}$ vector.
+        integer :: unit_
+        logical :: brief_
+        integer :: i, m
+        unit_  = merge(unit, output_unit, present(unit))
+        brief_ = merge(brief, .false., present(brief))
```

Comment:
In order to prevent users from accidentally using `disp` to print very long-dimensional arrays, causing unnecessary IO blockage, can we set brief to `.true.` by default? 
Previously, brief was set to `.false.` by default.

Reply:
I thought about the following implementation of the `brief` argument:
1. `brief_ = .true. .and. present(brief)`
5*5 (width * length elements)
2. `.not.present(brief)`
10*50
3. `brief_ = .false. .and. present(brief)`
all

Purpose: To prevent users from accidentally passing a very large array to disp, causing unnecessary io blockage.